drivers/spi/spi_ll_stm32.c

```diff
@@ -1514,18 +1530,33 @@ static int spi_stm32_pm_action(const struct device *dev,
 #endif /* CONFIG_PM_DEVICE */
 
 #ifdef CONFIG_SPI_STM32_INTERRUPT
-#define STM32_SPI_IRQ_HANDLER_DECL(id)					\
-	static void spi_stm32_irq_config_func_##id(const struct device *dev)
+#define STM32_SPI_IRQ_HANDLER_DECL(id)						\
+	static void spi_stm32_irq_config_func_##id(const struct device *dev);	\
+	IF_ENABLED(CONFIG_SOC_SERIES_STM32H7X,					\
+	(static void spi_stm32_irq_enable_func_##id(const struct device *dev);	\
+	static void spi_stm32_irq_disable_func_##id(const struct device *dev)))
 #define STM32_SPI_IRQ_HANDLER_FUNC(id)					\
-	.irq_config = spi_stm32_irq_config_func_##id,
+	.irq_config = spi_stm32_irq_config_func_##id,			\
+	IF_ENABLED(CONFIG_SOC_SERIES_STM32H7X,				\
+	(.irq_config_enable = spi_stm32_irq_enable_func_##id,		\
+	.irq_config_disable = spi_stm32_irq_disable_func_##id,))
 #define STM32_SPI_IRQ_HANDLER(id)					\
-static void spi_stm32_irq_config_func_##id(const struct device *dev)		\
+static void spi_stm32_irq_config_func_##id(const struct device *dev)	\
 {									\
 	IRQ_CONNECT(DT_INST_IRQN(id),					\
 		    DT_INST_IRQ(id, priority),				\
 		    spi_stm32_isr, DEVICE_DT_INST_GET(id), 0);		\
 	irq_enable(DT_INST_IRQN(id));					\
-}
+}									\
+	IF_ENABLED(CONFIG_SOC_SERIES_STM32H7X,				\
+(static void spi_stm32_irq_enable_func_##id(const struct device *dev)	\
+{									\
```

Comment:
`ARG_UNUSED(dev)`? (seem to be missing also right above).
Or maybe simply remove argument `dev`.

Reply:
Not necessary now that we directly `irq_enable/irq_disable` with the `irq_line` stored in config.